cyber/record/file/record_file_writer.h

```diff
@@ -123,7 +123,10 @@ bool RecordFileWriter::WriteSection(const T& message) {
     AERROR << "Do not support this template typename.";
     return false;
   }
-  Section section = {type, message.ByteSize()};
+  Section section;
+  /// zero out whole struct even if padded
+  memset(&section, 0, sizeof(section));
```

Comment:
Use the default initialization "s = {x, y}" directly is better, you don't need "memset".

Reply:
Later in the code
```
write(fd_, &section, sizeof(section));
```
writes the memory footprint of the object not just the object itself. There may be memory in the object between the members. Therefore, the program is reading from uninitialized memory. `memset` is an attempt at setting these bytes. `valgrind` reports an error, if this is not set. It's better to set this memory so that the record files are more regular.

Reply:
I think constructor construct "struct {int int}" with a block of contiguous memory, each field is located at a certain fixed offset from the start. 

But it may have its own reason for “valgrind” to report an error, it may have considered different compilers or otherwise.

I agree with this change.